csrc/cpu/fp8_utils.h

```diff
@@ -0,0 +1,256 @@
+#ifndef FP8_UTILS_H
+#define FP8_UTILS_H
+
+#include <immintrin.h>
+#include <cstdint>
+#include <cmath>
+#include <cstring>
+
+static inline __m512i _mm512_cvte5m2_fp16(__m256i a) {
+  return _mm512_slli_epi16(_mm512_cvtepi8_epi16(a), 8);
+}
+
+static inline __m256i _mm256_cvte5m2_fp16(__m128i a) {
```

Comment:
For all these function signatures that have just `e5m2` to signify `fp8e5m2`, I think it would be more clear to include the `fp8`